assets/js/blocks/cart-checkout/cart/edit.js

```diff
@@ -19,42 +18,35 @@ import EmptyCart from './empty-cart';
  * Component to handle edit mode of "Cart Block".
  */
 const CartEditor = ( { className } ) => {
-	const [ isFullCartMode, setFullCartMode ] = useState( true );
-
-	const toggleFullCartMode = () => {
-		setFullCartMode( ! isFullCartMode );
-	};
-
-	const getBlockControls = () => {
-		return (
-			<BlockControls className="wc-block-cart-toolbar">
-				<Toolbar>
-					<TextToolbarButton
-						onClick={ toggleFullCartMode }
-						isToggled={ isFullCartMode }
-					>
-						{ __( 'Full Cart', 'woo-gutenberg-products-block' ) }
-					</TextToolbarButton>
-					<TextToolbarButton
-						onClick={ toggleFullCartMode }
-						isToggled={ ! isFullCartMode }
-					>
-						{ __( 'Empty Cart', 'woo-gutenberg-products-block' ) }
-					</TextToolbarButton>
-				</Toolbar>
-			</BlockControls>
-		);
-	};
-
 	return (
 		<div className={ className }>
-			{ getBlockControls() }
-			{ isFullCartMode && (
-				<Disabled>
-					<FullCart />
-				</Disabled>
-			) }
-			<EmptyCart hidden={ isFullCartMode } />
+			<ViewSwitcher
+				label={ __( 'Edit', 'woo-gutenberg-products-block' ) }
+				views={ [
+					{
+						value: 'full',
+						name: __( 'Full Cart', 'woo-gutenberg-products-block' ),
+					},
+					{
+						value: 'empty',
+						name: __(
+							'Empty Cart',
+							'woo-gutenberg-products-block'
+						),
+					},
+				] }
+				defaultView={ 'full' }
+				render={ ( currentView ) => (
```

Comment:
Not sure if @senadir meant that, but when I read his suggestion I thought the render prop would be defined with each view individually.

So `views` would be in the form of 
```JSX
[
	{
		value: 'full',
		name: __( 'Full Cart', 'woo-gutenberg-products-block' ),
		render: () => (
			<Disabled>
				<FullCart />
			</Disabled>
		),
	},
	...
]
```

And `ViewSwitcher` would handle the logic to display one or the other internally.

Reply:
@senadir helped me here :) We need one of the views to always be rendered, but hidden, so that won't work for this case.

Reply:
@Aljullu that was my initial idea yeah, that and render props, but if we go that route, the empty state will not be rendered, see this comment on EmptyCart

```
// We need to always render `<InnerBlocks>` in the editor.
	// Otherwise, if the user saves the page without having
	// triggered the 'Empty Cart' view, inner blocks would not
	// be saved and they wouldn't be visible in the frontend.
	// We wrap them in a hidden `<div>` if the user is in
	// the editor 'Full Cart' view so they are not visible.
```

Reply:
Yeah, makes sense... Didn't see the `hidden` prop initially. :man_facepalming:  I still think it could be split and tied to each object in `views`, but as-is looks good to me as well, so no need to change that if it doesn't sound good. :slightly_smiling_face:

Reply:
Last thing @senadir @Aljullu Would it be preferable to rather than pass the render prop, to pass the views rendering in via `this.props.children`?

Reply:
I guess not, as it wouldn't have currentView context unlike the function.

Reply:
yes, you will be missing the currentView, render props are the use case for when you can’t use children